src/ScheduledTasks/Client/ScheduleClientImpl.cs

```diff
@@ -0,0 +1,275 @@
+
+using Microsoft.DurableTask.Client;
+using Microsoft.DurableTask.Client.Entities;
+using Microsoft.DurableTask.Entities;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DurableTask.ScheduledTasks;
+
+/// <summary>
+/// Represents a handle to a scheduled task, providing operations for managing the schedule.
+/// </summary>
+// TODO: Isolate system entity from user entities
+class ScheduleClientImpl : ScheduleClient
+{
+    readonly DurableTaskClient durableTaskClient;
+    readonly ILogger logger;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ScheduleClientImpl"/> class.
+    /// </summary>
+    /// <param name="client">The durable task client.</param>
+    /// <param name="scheduleId">The ID of the schedule.</param>
+    /// <param name="logger">The logger.</param>
+    /// <exception cref="ArgumentNullException">Thrown if <paramref name="client"/> or <paramref name="scheduleId"/> is null.</exception>
+    public ScheduleClientImpl(DurableTaskClient client, string scheduleId, ILogger logger)
+        : base(scheduleId)
+    {
+        this.durableTaskClient = client ?? throw new ArgumentNullException(nameof(client));
+        this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        this.EntityId = new EntityInstanceId(nameof(Schedule), this.ScheduleId);
+    }
+
+    /// <summary>
+    /// Gets the entity ID of the schedule.
+    /// </summary>
+    EntityInstanceId EntityId { get; }
+
+    /// <inheritdoc/>
+    public override async Task CreateAsync(ScheduleCreationOptions creationOptions, CancellationToken cancellation = default)
+    {
+        try
+        {
+            Check.NotNull(creationOptions, nameof(creationOptions));
+
+            ScheduleOperationRequest request = new ScheduleOperationRequest(this.EntityId, nameof(Schedule.CreateSchedule), creationOptions);
+            string instanceId = await this.durableTaskClient.ScheduleNewOrchestrationInstanceAsync(
+                new TaskName(nameof(ExecuteScheduleOperationOrchestrator)),
+                request,
+                cancellation);
+
+            // Wait for the orchestration to complete
+            OrchestrationMetadata state = await this.durableTaskClient.WaitForInstanceCompletionAsync(instanceId, true, cancellation);
+
+            if (state.RuntimeStatus != OrchestrationRuntimeStatus.Completed)
+            {
+                throw new InvalidOperationException($"Failed to create schedule '{this.ScheduleId}': {state.FailureDetails?.ErrorMessage ?? string.Empty}");
+            }
+        }
+        catch (OperationCanceledException) when (cancellation.IsCancellationRequested)
+        {
+            // the operation was cancelled as requested. No need to log this.
+            throw;
+        }
+        catch (Exception ex)
+        {
+            this.logger.ClientError(
+                nameof(this.CreateAsync),
+                this.ScheduleId,
+                ex);
```

Comment:
nit: make these log calls a single line. No need to wrap such short lines, creating more vertical scroll.